lib/commands/publish/npmUpdateAsLatest.js

```diff
@@ -12,13 +12,19 @@ module.exports = function npmUpdateAsLatest(changedPackages, versions, callback)
     return function (done) {
       while (true) {
         try {
-          logger.log("info", "Removing temporary npm dist-tag");
-          npmUtils.removeDistTag(pkg.name, "lerna-temp");
+          if (npmUtils.checkDistTag(pkg.name, "lerna-temp")) {
+            logger.log("info", "Removing temporary npm dist-tag");
+            npmUtils.removeDistTag(pkg.name, "lerna-temp");
+          }
+
           if (process.env.NPM_DIST_TAG) {
             logger.log("info", "Adding " + process.env.NPM_DIST_TAG + " npm dist-tag");
             npmUtils.addDistTag(pkg.name, versions[pkg.name], process.env.NPM_DIST_TAG);
+          } else if (flags.canary) {
+            logger.log("info", "Adding canary npm dist-tag");
+            npmUtils.addDistTag(pkg.name, pkg.pkg.version, "canary");
```

Comment:
If `pkg.pkg.version` has already been set, we can just use that everywhere instead of `versions[pkg.name]` right? In this case, the cross dependent versions are undefined (unless I should modify that instead)